libtiledbsoma/src/soma/column_buffer.h

```diff
@@ -136,6 +116,45 @@ class ColumnBuffer {
      */
     void attach(Query& query);
 
+    /**
+     * @brief Set the ColumnBuffer's data.
+     *
+     * @param data pointer to the beginning of the data to write
+     * @param num_elems the number of elements in the column
+     */
+    void set_data(
+        uint64_t num_elems,
+        const void* data,
+        uint64_t* offsets = nullptr,
+        uint8_t* validity = nullptr) {
+        num_cells_ = num_elems;
```

Comment:
num_cells does not seem be used here

Reply:
This sets `this->num_cells_` which gets used as `ColumnBuffer:size`.

Reply:
Thank you!